pandas/core/indexes/datetimelike.py

```diff
@@ -664,6 +665,12 @@ def __add__(self, other):
                 return self.shift(other)
             elif isinstance(other, (Index, datetime, np.datetime64)):
                 return self._add_datelike(other)
+            elif (isinstance(self, DatetimeIndex) and
+                  isinstance(other, np.ndarray) and other.size == 1 and
```

Comment:
huh? I believe we already check is_integer_dtype(other). this check is much too specific

Reply:
Under the status quo adding `np.array(1, dtype=np.int64)` to a `DatetimeIndex` behaves like adding `Timedelta(nanosecond=1)`.  `Series` raises (and a `Series` test fails unless this is caught here).

I'm open to ideas for how to make this less hacky.

Reply:
An alternative would be to -- for the time being -- check for this case in the Series op and avoid dispatching to `DatetimeIndex`.